### PR TITLE
feat: Add role icons

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Adds support for role icons.

**Reference Discord API Docs PRs or commits:**
- https://github.com/discord/discord-api-docs/pull/3847

### CODE REVIEW

File: rest/v9/guild.ts [hunk 1]
```diff
@@ -583,6 +595,14 @@ export interface RESTPatchAPIGuildRoleJSONBody {
 	 * Whether the role should be displayed separately in the sidebar
 	 */
 	hoist?: boolean | null;
+	/**
+	 * the role's icon image (if the guild has the `ROLE_ICONS` feature)
```
Comment (kyranet) [inline rest/v9/guild.ts:599]:
```suggestion
	 * The role's icon image (if the guild has the `ROLE_ICONS` feature)
```

File: rest/v9/guild.ts [hunk 2]
```diff
@@ -583,6 +595,14 @@ export interface RESTPatchAPIGuildRoleJSONBody {
...
+	 */
+	icon?: string | null;
+	/**
+	 * the role's unicode emoji (if the guild has the `ROLE_ICONS` feature)
```
Comment (kyranet) [inline rest/v9/guild.ts:603]:
```suggestion
	 * The role's unicode emoji (if the guild has the `ROLE_ICONS` feature)
```

File: rest/v9/guild.ts [hunk 3]
```diff
@@ -531,6 +531,18 @@ export interface RESTPostAPIGuildRoleJSONBody {
...
+	/**
+	 * the role's icon image (if the guild has the `ROLE_ICONS` feature)
+	 *
+	 * @default null
+	 */
+	icon?: string | null;
+	/**
+	 * the role's unicode emoji (if the guild has the `ROLE_ICONS` feature)
+	 *
+	 * @default null
```
Comment (kyranet) [inline rest/v9/guild.ts:543]:
```suggestion
	 * The role's unicode emoji (if the guild has the `ROLE_ICONS` feature)
```

File: rest/v9/guild.ts [hunk 4]
```diff
@@ -531,6 +531,18 @@ export interface RESTPostAPIGuildRoleJSONBody {
 	 * @default false
 	 */
 	hoist?: boolean | null;
+	/**
+	 * the role's icon image (if the guild has the `ROLE_ICONS` feature)
+	 *
+	 * @default null
```
Comment (kyranet) [inline rest/v9/guild.ts:537]:
```suggestion
	 * The role's icon image (if the guild has the `ROLE_ICONS` feature)
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: payloads/v9/guild.ts [hunk 5]
```diff
@@ -513,6 +513,10 @@ export const enum GuildFeature {
 	 * Guild has access to create private threads
 	 */
 	PrivateThreads = 'PRIVATE_THREADS',
+	/**
+	 * Guild is able to set role icons
+	 */
+	RoleIcons = 'ROLE_ICONS',
```
Comment (vladfrangu) [inline payloads/v9/guild.ts:519]: Please sort this alphabetically

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Also v8 this :)

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**